tests/metrics_test.go

```diff
@@ -167,87 +159,36 @@ var _ = Describe("Metrics", func() {
-				err := apiClient.List(ctx, rsList, client.InNamespace(sspDeploymentNamespace),
-					client.MatchingLabels{"name": "ssp-operator"})
-				Expect(err).ToNot(HaveOccurred())
-
-				for _, rs := range rsList.Items {
-					for _, env := range rs.Spec.Template.Spec.Containers[0].Env {
-						if env.Name == common.OperatorVersionKey &&
-							env.Value == version &&
-							*rs.Spec.Replicas > 0 &&
-							rs.Status.ReadyReplicas == *rs.Spec.Replicas {
-							return nil
-						}
-					}
-				}
-
-				return fmt.Errorf("operator version not yet updated")
-			}, env.ShortTimeout(), time.Second).Should(Succeed())
-		}
-
-		updateVersion := func(version string) {
-			Eventually(func() error {
-				deployment := getDeployment(sspDeploymentName, sspDeploymentNamespace)
-
-				envs := deployment.Spec.Template.Spec.Containers[0].Env
-
-				for i, env := range envs {
-					if env.Name == common.OperatorVersionKey {
-						envs[i].Value = version
-						break
-					}
-				}
-
-				return apiClient.Update(ctx, deployment)
-			}, env.ShortTimeout(), time.Second).ShouldNot(HaveOccurred())
-
-			waitForVersionUpdate(version)
-		}
+		var template *templatev1.Template
 
 		BeforeEach(func() {
-			strategy.SkipSspUpdateTestsIfNeeded()
-
-			By("saving original version")
-			originalVersion = ""
-			deployment := getDeployment(sspDeploymentName, sspDeploymentNamespace)
-			envs := deployment.Spec.Template.Spec.Containers[0].Env
-			for _, env := range envs {
-				if env.Name == common.OperatorVersionKey {
-					originalVersion = env.Value
-				}
-			}
-
 			waitUntilDeployed()
+			template = &getTemplates().Items[0]
 		})
 
-		AfterEach(func() {
-			updateVersion(originalVersion)
-			strategy.RevertToOriginalSspCr()
+		It("[test_id:TODO]should increment total_restored_common_templates during normal reconcile", func() {
+			skipIfUpgradeLane()
+
+			restoredCount := totalRestoredTemplatesCount()
+
+			template.Labels[common_templates.TemplateTypeLabel] = "test"
+			Expect(apiClient.Update(ctx, template)).To(Succeed())
+
+			Eventually(func() int {
+				return totalRestoredTemplatesCount()
+			}, 5*time.Minute, 10*time.Second).Should(Equal(restoredCount + 1))
 		})
 
 		It("[test_id:TODO]should not increment total_restored_common_templates during upgrades", func() {
-			testTemplate := createTestTemplate()
-
-			pauseSsp()
-			version := "test-" + rand.String(5)
-			updateVersion(version)
+			restoredCount := totalRestoredTemplatesCount()
 
-			template := &templatev1.Template{}
-			Expect(apiClient.Get(ctx, testTemplate.GetKey(), template)).To(Succeed())
-			testTemplate.Update(template)
+			template.Labels[common_templates.TemplateTypeLabel] = "test"
+			template.Labels[common_templates.TemplateVersionLabel] = "v" + rand.String(5)
 			Expect(apiClient.Update(ctx, template)).To(Succeed())
 
-			unpauseSsp()
-			waitUntilDeployed()
-
-			newRestoredCount := totalRestoredTemplatesCount()
-			Expect(newRestoredCount).To(BeZero())
+			Consistently(func() int {
```

Comment:
Using `Consistently()` is not ideal. There may be a way to wait for the template update directly, and not just for some arbitrary duration.

Can you add a `// TODO:` comment here, so that we don't forget about it?

Reply:
added